parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -103,18 +103,9 @@ logs_to_events "AKS.CSE.disableSystemdResolved" disableSystemdResolved
 
 logs_to_events "AKS.CSE.configureAdminUser" configureAdminUser
 
-VHD_LOGS_FILEPATH=/opt/azure/vhd-install.complete
-if [ -f $VHD_LOGS_FILEPATH ]; then
-    echo "detected golden image pre-install"
-    logs_to_events "AKS.CSE.cleanUpContainerImages" cleanUpContainerImages
-    FULL_INSTALL_REQUIRED=false
-else
-    if [[ "${IS_VHD}" = true ]]; then
-        echo "Using VHD distro but file $VHD_LOGS_FILEPATH not found"
-        exit $ERR_VHD_FILE_NOT_FOUND
-    fi
-    FULL_INSTALL_REQUIRED=true
-fi
+export -f should_skip_binary_cleanup
+SKIP_BINARY_CLEANUP=$(retrycmd_if_failure_no_stats 10 1 10 bash -cx should_skip_binary_cleanup)
+FULL_INSTALL_REQUIRED=$(handleContainerImageCleanup $SKIP_BINARY_CLEANUP $IS_VHD)
```

Comment:
I believe `set -e` will be configured at this point, though I think we should explicitly check `$?` for the `exit $ERR_VHD_FILE_NOT_FOUND` case since we're calling the func in a subshell, otherwise if `set -e` isn't set the exit statement will essentially do nothing

we actually do the same thing up above on lines 90-96 when checking `exit should_skip_nvidia_drivers `